astroquery/xmm_newton/core.py

```diff
@@ -192,13 +201,19 @@ def get_postcard(self, observation_id, image_type="OBS_EPIC",
                   'OBS_IMAGE_TYPE': image_type,
                   'PROTOCOL': 'HTTP'}
 
-        response = self._request('GET', self.data_url, params, cache=True)
-        response.raise_for_status()
         if verbose:
             log.info(self.data_url)
 
-        if response is not None:
+        response = None
+        try:
+            response = self._request('GET', self.data_url, params, cache=True)
+            response.raise_for_status()
+        except HTTPError:
+            response = None
 
+        if response is None:
+            log.error("Unable to access URL {0}".format(self.data_url))
+        else:
```

Comment:
So it seems here the purpose is to check whether the request returns a good response, and if it doesn't, do something else.

Would it be possible to put the 'something else' response in the `except ...` block here?  I'm not sure exactly what the fail case is supposed to be, but if the URL fails to load, we usually want to alert the user that the request failed and issue a traceback.  If there's a reason to do something different and handle a failed URL internally, could you comment (in the code) about why that's needed?